src/composable/types/Twap.ts

```diff
@@ -18,84 +19,131 @@ export const MAX_UINT32 = BigNumber.from(2).pow(32).sub(1) // 2^32 - 1
+ */
+export type TwapPart = {
+  /**
+   * Amount of sellToken to sell in each part
+   */
+  partSellAmount: BigNumber
+
+  /**
+   * Minimum amount of buyToken that must be bought across the entire TWAP
+   */
+  minPartLimit: BigNumber
+}
+
 /**
  * Parameters for a TWAP order, as expected by the contract's `staticInput`.
  */
-export type TWAPData = {
-  // which token to sell
+export type TwapData = {
+  /**
+   * which token to sell
+   */
   readonly sellToken: string
-  // which token to buy
+
+  /**
+   * which token to buy
+   */
   readonly buyToken: string
-  // who to send the tokens to
+
+  /**
+   * who to send the tokens to
+   */
   readonly receiver: string
-  // amount of sellToken to sell in each part
+
+  /**
+   * amount of sellToken to sell in each part
+   */
   readonly partSellAmount: BigNumber
-  // minimum amount of buyToken that must be bought in each part
+
+  /**
+   * minimum amount of buyToken that must be bought in each part
+   */
   readonly minPartLimit: BigNumber
-  // start time of the TWAP
+
+  /**
+   * start time of the TWAP
+   */
   readonly t0: BigNumber
-  // number of parts
+
+  /**
+   * number of parts
+   */
   readonly n: BigNumber
-  // duration of the TWAP interval
+
+  /**
+   * duration of the TWAP interval
+   */
   readonly t: BigNumber
-  // whether the TWAP is valid for the entire interval or not
+
+  /**
+   * whether the TWAP is valid for the entire interval or not
+   */
   readonly span: BigNumber
-  // appData
+
+  /**
+   * Meta-data associated with the order. Normally would be the keccak256 hash of the document generated in http://github.com/cowprotocol/app-data
+   *
+   * This hash should have been uploaded to the API https://api.cow.fi/docs/#/default/put_api_v1_app_data__app_data_hash_ and potentially to other data availability protocols like IPFS.
+   *
+   */
   readonly appData: string
 }
 
 /**
  * `ComposableCoW` implementation of a TWAP order.
  * @author mfw78 <mfw78@rndlabs.xyz>
  */
-export class TWAP extends BaseConditionalOrder<TWAPData, TWAPDataParams> {
+export class Twap extends ConditionalOrder<TwapData, TwapDataParams> {
   /**
-   * @see {@link BaseConditionalOrder.constructor}
+   * @see {@link ConditionalOrder.constructor}
    * @throws If the TWAP order is invalid.
    * @throws If the TWAP order is not ABI-encodable.
    * @throws If the handler is not the TWAP address.
    */
-  constructor(
-    handler: string,
-    salt: string = utils.keccak256(ethers.utils.randomBytes(32)),
-    staticInput: TWAPDataParams,
-    hasOffChainInput = false
-  ) {
+  constructor(params: ConditionalOrderArguments<TwapDataParams>) {
+    const { handler, salt, staticInput, hasOffChainInput } = params
     // First, verify that the handler is the TWAP address
-    if (handler !== TWAP_ADDRESS) throw new Error('InvalidHandler')
+    if (handler !== TWAP_ADDRESS) throw new Error(`InvalidHandler: Expected: ${TWAP_ADDRESS}, provided: ${handler}`)
```

Comment:
Added some additional context (and adapted the test)

Reply:
Makes sense. I would suggest to do a refactor once we have more than one impl?

Reply:
Sure, #148 refers.